### ISSUE TITLE
False positive hardcoded-password-string (S105)

### ISSUE THREAD

**Issue by rossnomann** [astral-sh/ruff#7816]
A minimal code snippet that reproduces the bug:

```python
import enum


class Method(str, enum.Enum):
    token_get = "token_get"
    token_refresh = "token_refresh"

```

The current Ruff settings (`pyproject.toml`):
```toml
[tool.ruff]
line-length = 120
ignore = [
  "ANN101",  # missing-type-self
  "ANN102",  # missing-type-cls
  "D",  # pydocstyle
  "FIX",  #  flake8-fixme
  "TD003"  # Missing issue link
]
select = ["ALL"]
target-version = "py311"
```
Ruff version: 0.0.292

**Comment by charliermarsh** [comment]
Bandit flags these too. Could you instead use `enum.auto()` for the values? IIUC that has the same effect, and avoids the hardcoded constant.

**Comment by rossnomann** [comment]
Yes, `enum.auto` works well, thanks
